usecases/modules/vectorizer.go

```diff
@@ -202,14 +202,22 @@ func (p *Provider) vectorize(ctx context.Context, object *models.Object, class *
 
 	if vectorizer, ok := found.(modulecapabilities.Vectorizer); ok {
 		if p.shouldVectorizeObject(object, cfg) {
-			comp, err := compFactory()
-			if err != nil {
-				return fmt.Errorf("failed creating properties comparator: %w", err)
+			var targetProperties []string = nil
+			vecConfig, ok := modConfig[found.Name()]
+			if ok {
+				if properties, ok := vecConfig.(map[string]interface{})["properties"]; ok {
+					targetProperties = properties.([]string)
+				}
 			}
-			vector, additionalProperties, err := vectorizer.VectorizeObject(ctx, object, comp, cfg)
-			if err != nil {
-				return fmt.Errorf("update vector: %w", err)
+			needsRevectorization, additionalProperties, vector := reVectorize(ctx, cfg, vectorizer, object, class, targetProperties, targetVector, findObjectFn)
+			if needsRevectorization {
```

Comment:
what happens if target vector is explicitly given with object. is vectorization skipped in that case?

Reply:
`shouldVectorize`would skip that object and it would not reach this code:

```
	vectorizer := p.getVectorizer(class, targetVector)
	if vectorizer == config.VectorizerModuleNone {
		vector := p.getVector(object, targetVector)
		if hnswConfig.Skip && len(vector) > 0 {
			logger.WithField("className", class.Class).
				Warningf(warningSkipVectorProvided)
		}

		return false, nil
	}
```

Reply:
right, missed that.

BTW: `shouldVectorize`, `shouldVectorizeObject`, `reVectorize` could be unified/merged at some point